apph-back/src/main/java/com/viseo/apph/service/TagService.java

```diff
@@ -0,0 +1,58 @@
+package com.viseo.apph.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.viseo.apph.dao.TagDAO;
+import com.viseo.apph.dao.UserDAO;
+import com.viseo.apph.domain.Tag;
+import com.viseo.apph.domain.User;
+import io.jsonwebtoken.Claims;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+public class TagService {
+    @Autowired
+    UserDAO userDao;
+
+    @Autowired
+    TagDAO tagDAO;
+
+    @Transactional
+    public List<Tag> getTags(Claims claims) {
+        User user = userDao.getUserByLogin(claims.get("login").toString());
+        return tagDAO.getTagsByUser(user.getId());
+    }
+
+    @Transactional
+    public Tag createTag(String tagName, Claims claims) {
```

Comment:
je vois que cette méthode n'est jamais utilisée, autant la supprimer. On peut considérer que créer un seul tag c'est finalement itérer sur une liste à un seul élément.
Sinon (à toi de voir) tu pourrais appeler cette méthode pour chaque tag à créer dans `createListTags`...